### PR TITLE
Add zuulv3 work

### PR DESCRIPTION
**Pull request by jamielennox**
This is a combination of fixes that make zuulv3 work in my environment.
It is not everything and not ready for production but something we can
start working with.

Signed-off-by: Jamie Lennox <jamielennox@gmail.com>

### CODE REVIEW

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/check_github/BonnyCI/hoist/255/1490135301.24/bonnyci-run-check-multinode : FAILURE in 3m 59s

File: roles/zuul/templates/etc/zuul/zuul.conf [hunk 1]
```diff
@@ -19,24 +19,37 @@ start = {{ zuul_gearman_server_start }}
...
 {% endfor %}
 
 {% endfor %}
+{% endif %}
 
 [zuul]
+{% if not zuul_zuul_v3 %}
 layout_config = /etc/zuul/config/layout.yaml
+{% endif %}
 log_config = /etc/zuul/server-logging.conf
```
Comment (gandelman-a) [inline roles/zuul/templates/etc/zuul/zuul.conf]: the logging config ends up being installed as either server-* or scheduler-* depending on version, so this needs to updated to set it accordingly.

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/check_github/BonnyCI/hoist/255/1490137373.65/bonnyci-run-check-multinode : FAILURE in 3m 59s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/check_github/BonnyCI/hoist/255/1490138496.92/bonnyci-run-check-multinode : FAILURE in 1m 14s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/check_github/BonnyCI/hoist/255/1490141640.29/bonnyci-run-check-multinode : FAILURE in 4m 46s

**Comment by jamielennox** [comment]
recheck

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/check_github/BonnyCI/hoist/255/1490142426.78/bonnyci-run-check-multinode : FAILURE in 2m 16s

**Comment by jamielennox** [comment]
recheck

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/check_github/BonnyCI/hoist/255/1490142783.38/bonnyci-run-check-multinode : FAILURE in 2m 15s

**Comment by jamielennox** [comment]
recheck

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/check_github/BonnyCI/hoist/255/1490150722.5/bonnyci-run-check-multinode : FAILURE in 2m 37s

**Comment by jamielennox** [comment]
recheck

File: roles/zuul/templates/etc/zuul/zuul.conf [hunk 2]
```diff
@@ -19,24 +19,38 @@ start = {{ zuul_gearman_server_start }}
...
+{% if zuul_zuul_v3 %}
+log_config = /etc/zuul/scheduler-logging.conf
+pidfile = /var/run/zuul-scheduler/zuul-sscheduler.pid
+tenant_config = /etc/zuul/tenant.yaml
+zookeeper_hosts = {{ zuul_zookeeper_hosts }}
+
+[executor]
+log_config = /etc/zuul/launcher-logging.conf
+
+[launcher]
```
Comment (gandelman-a) [inline roles/zuul/templates/etc/zuul/zuul.conf:44]: Is this [launcher] heading still stupposed to be here when we're deploying v3?

File: roles/zuul/templates/etc/zuul/zuul.conf [hunk 3]
```diff
@@ -19,24 +19,38 @@ start = {{ zuul_gearman_server_start }}
...
 
 {% endfor %}
+{% endif %}
 
 [zuul]
+state_dir = /var/lib/zuul
+url_pattern = {{ zuul_logs_url }}/{build.parameters[LOG_PATH]}
+{% if zuul_zuul_v3 %}
+log_config = /etc/zuul/scheduler-logging.conf
+pidfile = /var/run/zuul-scheduler/zuul-sscheduler.pid
```
Comment (gandelman-a) [inline roles/zuul/templates/etc/zuul/zuul.conf]: typo: sscheduler

File: roles/zuul/templates/etc/zuul/zuul.conf (hunk 2 above)
Comment (jamielennox) [reply roles/zuul/templates/etc/zuul/zuul.conf:44]: yea, i'm assuming it's still a transitional thing but there are still a couple of variables that get pulled from launcher

File: roles/zuul/templates/etc/zuul/zuul.conf (hunk 3 above)
Comment (jamielennox) [reply roles/zuul/templates/etc/zuul/zuul.conf]: damn, - i'm actually pretty sure this gets ignored when we don't use the daemon mode anyway.

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/check_github/BonnyCI/hoist/255/1490225613.5/bonnyci-run-check-multinode : FAILURE in 3m 44s

**Comment by gandelman-a** [comment]
recheck

**Review by gandelman-a — APPROVED** [review]

**Pull request merged**